[stable/datadog] add podLabels option

max-rocket-internet opened the pull request:
We use pod labels in prometheus and also in alertsmanager so would be great to have this available in this chart.


#### What this PR does / why we need it:

Add optional pod labels.

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

max-rocket-internet committed:
[stable/datadog] add podLabels option

We use pod labels in prometheus and also in alertsmanager so would be great to have this available in this chart.

Signed-off-by: Max Williams <max.williams@deliveryhero.com>

hkaj commented on stable/datadog/README.md:
```diff
@@ -219,13 +219,14 @@ helm install --name <RELEASE_NAME> \
 | `fullnameOverride`                       | Override full name of app                                                                 | `nil`                                       |
 | `rbac.create`                            | If true, create & use RBAC resources                                                      | `true`                                      |
 | `rbac.serviceAccount`                    | existing ServiceAccount to use (ignored if rbac.create=true)                              | `default`                                   |
+| `podLabels`                              | labels to add to each pod                                                                 | `{}`                                        |
```
```suggestion
| `podLabels`                              | labels to add to each pod                                                                 | `nil`                                        |
```

max-rocket-internet replied:
OK sure but it's not consistent with other charts. If the value is a hash then `{}` is normally the default. Even in the README now there's `{}` for `deployment.affinity`

hkaj commented on stable/datadog/templates/daemonset.yaml:
```diff
@@ -14,6 +14,9 @@ spec:
         app: {{ template "datadog.fullname" . }}
+{{- if .Values.podLabels }}
+{{ toYaml .Values.podLabels | indent 8 }}
+{{- end }}
```
```suggestion
        {{- end }}
```

hkaj commented on stable/datadog/templates/daemonset.yaml:
```diff
@@ -14,6 +14,9 @@ spec:
     metadata:
       labels:
         app: {{ template "datadog.fullname" . }}
+{{- if .Values.podLabels }}
```
```suggestion
        {{- if .Values.podLabels }}
```

hkaj commented on stable/datadog/values.yaml:
```diff
@@ -360,6 +360,7 @@ rbac:
   serviceAccountName: default
 
 tolerations: []
+podLabels: {}
```
Could you move it under the `daemonset` section please? We're in the process of deprecating the deployment, in favor of `clusterchecksDeployment`, so it's on feature freeze.

max-rocket-internet replied:
Done

hkaj commented on stable/datadog/templates/deployment.yaml:
```diff
@@ -16,6 +16,9 @@ spec:
         type: deployment
+{{- if .Values.podLabels }}
+{{ toYaml .Values.podLabels | indent 8 }}
+{{- end }}
```
Could you please remove them from the deployment? It's on feature freeze and will be deprecated in a few versions.

You can add it to the [clusterchecks deployment](https://github.com/helm/charts/blob/master/stable/datadog/templates/agent-clusterchecks-deployment.yaml) if you want, maybe by splitting the setting to have `.Values.clusterchecksDeployment.podLabels` and `.Values.daemonset.podLabels`.

max-rocket-internet replied:
OK removed from the deployment. I've left the clusterchecks thing as I don't know what it is.

max-rocket-internet committed:
updates after review

- Changing indentation of condition
- Removing pod labels from deployment
- Changing default in readme to "nil"
- Moving podLabels value inside daemonset map

Signed-off-by: Max Williams <max.williams@deliveryhero.com>

hkaj commented on stable/datadog/README.md:
```diff
@@ -219,13 +219,14 @@ helm install --name <RELEASE_NAME> \
 | `fullnameOverride`                       | Override full name of app                                                                 | `nil`                                       |
 | `rbac.create`                            | If true, create & use RBAC resources                                                      | `true`                                      |
 | `rbac.serviceAccount`                    | existing ServiceAccount to use (ignored if rbac.create=true)                              | `default`                                   |
+| `podLabels`                              | labels to add to each pod                                                                 | `nil`                                       |
```
let's group this option with other `daemonset` options, around L260? And prefix it with `daemonset.`

max-rocket-internet replied:
OK done!

max-rocket-internet committed:
Fix readme

Signed-off-by: Max Williams <max.williams@deliveryhero.com>
